meson.build

```diff
@@ -0,0 +1,111 @@
+project('opensc', 'c',
+    version: '0.23.165'
+)
+
+cc = meson.get_compiler('c')
+cdata = configuration_data()
+check_headers = [
+    'dlfcn.h',
+    'fcntl.h',
+    'getopt.h',
+    'pcsclite.h',
+    'strings.h',
+    'string.h',
+    'sys/endian.h',
+    'sys/mman.h',
+    'sys/stat.h',
+    'sys/time.h',
+    'sys/types.h',
+    'sys/wait.h',
+    'unistd.h',
+    'wcautil.h',
+    'winscard.h',
+    'stdint.h',
+]
+
+add_project_arguments('-DHAVE_CONFIG_H', language: 'c')
+
+foreach h : check_headers
+  if cc.has_header(h)
+    cdata.set('HAVE_' + h.underscorify().to_upper(), 1)
+  endif
+endforeach
+
+check_functions = [
+#   check token ['HAVE_DECL_STRLCAT']
+#   check token ['HAVE_DECL_STRLCPY']
+#   check token ['HAVE_DOPRNT']
+#   check token ['HAVE_EAC_OBJ_NID2OBJ']
+#   check token ['HAVE_EXPLICIT_BZERO']
+    ['HAVE_GETLINE', 'getline', '#include<stdio.h>'],
+    ['HAVE_GETOPT_LONG', 'getopt_long', '#include<getopt.h>'],
+#   check token ['HAVE_GETPASS']
+    ['HAVE_GETTIMEOFDAY', 'gettimeofday', '#include<sys/time.h>'],
+    ['HAVE_MEMSET', 'memset', '#include<string.h>'],
+#   check token ['HAVE_MEMSET_S']
+#   check token ['HAVE_MKDIR']
+    ['HAVE_PTHREAD', 'pthread_create', '#include<pthread.h>'],
+#   check token ['HAVE_PTHREAD_PRIO_INHERIT']
+    ['HAVE_SIGACTION', 'sigaction', '#include<signal.h>'],
+#   check token ['HAVE_STAT_EMPTY_STRING_BUG']
+    ['HAVE_STRDUP', 'strdup', '#include<string.h>'],
+    ['HAVE_STRERROR', 'strerror', '#include<string.h>'],
+#   check token ['HAVE_STRNLEN']
+#   check token ['HAVE_VPRINTF']
+    ['HAVE_STRERROR', 'strerror', '#include<string.h>'],
+    ['HAVE_BUILTIN_OVERFLOW', '__builtin_uadd_overflow', ''],
+]
+
+foreach f : check_functions
+  if cc.has_function(f.get(1), prefix : f.get(2))
+    cdata.set(f.get(0), 1)
+  endif
+endforeach
+
+cdata.set_quoted('SC_PKCS15_PROFILE_DIRECTORY', get_option('prefix') / get_option('datadir') / 'opensc')
+cdata.set('SIZEOF_VOID_P', cc.sizeof('void*'))
+
+thread_dep = dependency('threads', required: false)
+zlib_dep = dependency('zlib', required: false)
+openssl_dep = dependency('openssl', required: false)
+readline_dep = dependency('readline', required: false)
+dl_dep = cc.find_library('dl', required : false)
+openpace_dep = dependency('libeac', required: false)
+
+cdata.set('ENABLE_READLINE', readline_dep.found())
+cdata.set('ENABLE_ZLIB', zlib_dep.found())
+cdata.set('HAVE_ZLIB_H', zlib_dep.found())
+cdata.set('ENABLE_OPENSSL', openssl_dep.found())
+cdata.set('HAVE_OPENSSL_CRYPTO_H', openssl_dep.found())
+cdata.set('ENABLE_OPENPACE', openpace_dep.found())
+cdata.set('ENABLE_SM', get_option('sm').enabled())
```

Comment:
The type of `sm` is a boolean so this should probably be `cdata.set('ENABLE_SM', get_option('sm'))`. The `enabled` is used when the type of the option is `feature`.